homeassistant/components/paperless_ngx/config_flow.py

```diff
@@ -76,3 +68,68 @@ async def async_step_user(
         return self.async_show_form(
             step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
         )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Reconfigure flow for Paperless-ngx integration."""
+        return await self.async_step_reauth_confirm(user_input)
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that informs the user that reauth is required."""
+
+        reauth_entry = (
+            self._get_reauth_entry()
+            if self.source == SOURCE_REAUTH
+            else self._get_reconfigure_entry()
+        )
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                await self._validate_input(user_input)
+            except PaperlessConnectionError:
+                errors[CONF_URL] = "cannot_connect"
+            except PaperlessInvalidTokenError:
+                errors[CONF_API_KEY] = "invalid_api_key"
+            except PaperlessInactiveOrDeletedError:
+                errors[CONF_API_KEY] = "user_inactive_or_deleted"
+            except PaperlessForbiddenError:
+                errors[CONF_API_KEY] = "forbidden"
+            except InitializationError:
+                errors[CONF_URL] = "cannot_connect"
+            except Exception as err:  # noqa: BLE001
+                LOGGER.exception("Unexpected exception: %s", err)
+                errors["base"] = "unknown"
+            else:
+                return self.async_update_reload_and_abort(reauth_entry, data=user_input)
+
+        return self.async_show_form(
+            step_id="reauth_confirm" if self.source == SOURCE_REAUTH else "reconfigure",
+            data_schema=self.add_suggested_values_to_schema(
+                data_schema=STEP_USER_DATA_SCHEMA,
+                suggested_values={
+                    CONF_URL: user_input[CONF_URL]
+                    if user_input is not None
+                    else reauth_entry.data[CONF_URL],
+                },
+            ),
```

Comment:
reconfigure -> change both url and api key
reauth -> only api key

So I don't think you can merge the 2 steps nicely